File: lib/reels/reel_model.dart
class Reel {
  final String username;
  final String caption;
  final String imageUrl;
  int likes;
  bool liked;

  Reel({
    required this.username,
    required this.caption,
    required this.imageUrl,
    this.likes = 0,
    this.liked = false,
  });
}

File: backend/src/models/reel.model.js
import mongoose from "mongoose";

const ReelSchema = new mongoose.Schema(
  {
    user: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },

    videoUrl: { type: String, required: true },
    caption: { type: String, default: "" },

    likes: [{ type: mongoose.Schema.Types.ObjectId, ref: "User" }],
    saves: [{ type: mongoose.Schema.Types.ObjectId, ref: "User" }],
    views: { type: Number, default: 0 },
    comments: [{
      user: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
      text: { type: String, required: true },
      createdAt: { type: Date, default: Date.now },
    }],
  },
  { timestamps: true }
);

export const Reel = mongoose.model("Reel", ReelSchema);
